business_requirement_deliverable_cost/models/business.py

```diff
@@ -0,0 +1,139 @@
+# -*- coding: utf-8 -*-
+# © 2016 Elico Corp (https://www.elico-corp.com).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openerp import api, fields, models
+
+
+class BusinessRequirementResource(models.Model):
+    _inherit = "business.requirement.resource"
+
+    sale_price_unit = fields.Float(
+        string='Sales Price',
+        readonly=True,
+        groups='business_requirement_deliverable_cost.\
+group_business_requirement_estimation',
+    )
+    sale_price_total = fields.Float(
+        compute='_compute_sale_price_total',
+        string='Total Revenue',
+        groups='business_requirement_deliverable_cost.\
+group_business_requirement_estimation',
+    )
+    unit_price = fields.Float(
+        groups='business_requirement_deliverable_cost.\
+group_business_requirement_cost_control',
+    )
+    price_total = fields.Float(
+        groups='business_requirement_deliverable_cost.\
+group_business_requirement_cost_control',
+    )
+
+    @api.multi
+    @api.depends('sale_price_unit', 'qty')
+    def _compute_sale_price_total(self):
+        for resource in self:
+            resource.sale_price_total = resource.sale_price_unit * resource.qty
```

Comment:
But what about the conversion with UoM factor?
Eric CAUDAL --
*Eric Caudal* /[Founder and CEO]/
Skype: elico.corp. Phone: + 86 186 2136 1670 (Cell), + 86 21 6211 
8017/27/37 (Office)
*Elico Shanghai* (Hong Kong/Shenzhen/Singapore) 
<http://www.elico-corp.com> /*Odoo Gold Partner, best Odoo Partner 2014 
for APAC*/
On 03/08/2016 06:19 PM, Luke Zheng wrote:
>
> In business_requirement_deliverable_cost/models/business.py 
> <https://github.com/xie8899/project/pull/16#discussion_r55339940>:
>
> > +group_business_requirement_estimation',
> > +    )
> > +    unit_price = fields.Float(
> > +        groups='business_requirement_deliverable_cost.\
> > +group_business_requirement_cost_control',
> > +    )
> > +    price_total = fields.Float(
> > +        groups='business_requirement_deliverable_cost.\
> > +group_business_requirement_cost_control',
> > +    )
> > +
> > +    @api.multi
> > +    @api.depends('sale_price_unit', 'qty')
> > +    def _compute_sale_price_total(self):
> > +        for resource in self:
> > +            resource.sale_price_total = resource.sale_price_unit * resource.qty
>
> the sale_price_unit will recompute when uom change
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/xie8899/project/pull/16/files#r55339940>.
>